spec/support/queries/filters/shared_filter_examples.rb

```diff
@@ -370,6 +370,54 @@
   end
 end
 
+RSpec.shared_examples_for "list_contains query filter" do
+  include_context "filter tests"
+  let(:type) { :list_contains }
+
+  describe "#scope" do
+    let(:values) { valid_values }
+
+    context 'for "~"' do
+      let(:operator) { "~" }
+
+      it "is the same as handwriting the query" do
+        handwritten_scope_sql = <<-SQL.squish
+          SELECT "projects".* FROM "projects"
+            INNER JOIN "project_custom_field_project_mappings" ON "project_custom_field_project_mappings"."project_id" = "projects"."id"
+            WHERE "project_custom_field_project_mappings"."custom_field_id" IN (#{values.join(', ')})
+        SQL
+
+        expect(instance.scope.to_sql).to eql handwritten_scope_sql
+      end
+    end
```

Comment:
I have in the past written those SQL based expectations myself. Nowadays I consider them too brittle as e.g. the order might change between AR versions without the behaviour changing. 

The spec here is also specific for the `project_custom_field_project_mappings`

Reply:
Fine with me.